src/test/java/com/notably/logic/suggestion/commands/DeleteSuggestionCommandTest.java

```diff
@@ -0,0 +1,132 @@
+import com.notably.model.block.BlockModelImpl;
+import com.notably.model.block.Title;
+import com.notably.model.suggestion.SuggestionItem;
+import com.notably.model.suggestion.SuggestionItemImpl;
+import com.notably.model.suggestion.SuggestionModel;
+import com.notably.model.suggestion.SuggestionModelImpl;
+import com.notably.model.viewstate.ViewStateModel;
+import com.notably.model.viewstate.ViewStateModelImpl;
+
+public class DeleteSuggestionCommandTest {
+    private static AbsolutePath toRoot;
+    private static AbsolutePath toCs2103;
+    private static AbsolutePath toCs3230;
+    private static AbsolutePath toCs2103Week1;
+    private static AbsolutePath toCs2103Week2;
+    private static AbsolutePath toCs2103Week3;
+    private static AbsolutePath toCs2103Week1Lecture;
+    private static Model model;
+
+    @BeforeAll
+    public static void setUp() throws InvalidPathException {
+        // Set up paths
+        toRoot = AbsolutePath.fromString("/");
+        toCs2103 = AbsolutePath.fromString("/CS2103");
+        toCs3230 = AbsolutePath.fromString("/CS3230");
+        toCs2103Week1 = AbsolutePath.fromString("/CS2103/Week1");
+        toCs2103Week2 = AbsolutePath.fromString("/CS2103/Week2");
+        toCs2103Week3 = AbsolutePath.fromString("/CS2103/Week3");
+        toCs2103Week1Lecture = AbsolutePath.fromString("/CS2103/Week1/Lecture");
+
+        // Set up model
+        BlockModel blockModel = new BlockModelImpl();
+        SuggestionModel suggestionModel = new SuggestionModelImpl();
+        ViewStateModel viewStateModel = new ViewStateModelImpl();
+        model = new ModelManager(blockModel, suggestionModel, viewStateModel);
+
+        // Add test data to model
+        Block cs2103 = new BlockImpl(new Title("CS2103"));
+        Block cs3230 = new BlockImpl(new Title("CS3230"));
+        model.addBlockToCurrentPath(cs2103);
+        model.addBlockToCurrentPath(cs3230);
+
+        Block week1 = new BlockImpl(new Title("Week1"));
+        Block week2 = new BlockImpl(new Title("Week2"));
+        Block week3 = new BlockImpl(new Title("Week3"));
+        model.setCurrentlyOpenBlock(toCs2103);
+        model.addBlockToCurrentPath(week1);
+        model.addBlockToCurrentPath(week2);
+        model.addBlockToCurrentPath(week3);
+
+        Block lecture = new BlockImpl(new Title("Lecture"));
+        model.setCurrentlyOpenBlock(toCs2103Week1);
+        model.addBlockToCurrentPath(lecture);
+    }
+
+    @Test
+    public void constructor_nullPath_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DeleteSuggestionCommand(null));
+    }
+
+    @Test
+    public void execute_nullModel_throwsNullPointerException() {
+        DeleteSuggestionCommand deleteSuggestionCommand = new DeleteSuggestionCommand(toRoot);
+        assertThrows(NullPointerException.class, () -> deleteSuggestionCommand.execute(null));
+    }
+
+    @Test
+    public void execute_displayText() {
```

Comment:
Actually this should be called `execute_generateResponseCorrectly` or something else that is more general? `displayText` is just a subset of what you’re testing here

Reply:
alright!